lang/en/docs/software-directory/machine-learning/python-ml/components.md

```diff
@@ -16,21 +16,216 @@ available executables and flavors under the
 PythonML is based on the `python` [executable](../../../software/components.md#executables), and through this executable
 the implemented ML calculations can be performed.
 
+## Workflow Structure
+
+PythonML workflows contain two subworkflows. The first, called "Set Up the Job" performs actions such as copying data
+and setting environment variables necessary for the job to run. The second, called "Machine Learning," contains the
+actual machine learning units. A diagram can be found below, based on the bank workflow located
+[here](https://platform.exabyte.io/analytics/workflows/rg3HcXd9Rg7hTa5Q3).
+
+![Workflow Diagram](../../../images/software-directory/machine-learning/python-ml/ML_Train_Diagram.png)
+
+### Subworkflow: Set Up the Job
+
+This subworkflow facilitates setting up the PythonML job. Users should not need to edit this workflow. The configuration
+of this workflow is handled automatically on the
+when [the predict workflow is generated](../../../properties-directory/non-scalar/workflow.md).
+
+### Subworkflow: Machine Learning
+
+This subworkflow is where a user's requested machine learning units reside. This subworkflow is generally the one that
+users are expected to modify, to add or remove different machine learning workflow units. In the above diagram, we find
+the following units:
+
+| Unit Name                     | Flavor                                       | Description |
+|-------------------------------|----------------------------------------------|-------------|
+| `Setup Packages and Variables`| `pyml:setup_variables_packages`              |Contains functions and configuration essential for all python-ML Workflows|
+| ` Data Input`                 | `pyml:data_input:read_csv:pandas`            |for reading in CSV data using Pandas [^1]|
+| `Data Standardize`            | `pyml:pre_processing:standardization:sklearn`|scales the data [^2] such that it has a mean of 0 and a standard deviation of 1, as implemented in Scikit-Learn [^3]|
+| `Model Train and Predict`     | `pyml:model:multilayer_perceptron:sklearn`   |a Multilayer Perceptron [^4] implemented in Scikit-Learn [^5]|
+| `Parity Plot`                 | `pyml:post_processing:parity_plot:matplotlib`|generates a parity plot [^6] using Matplotlib [^7]|
+
 ## Flavors
 
-The following flavors are available within the current ML implementation:
+A variety of flavors are available for use in machine learning. Generally, flavors behave different based on whether a
```

Comment:
Do we ever explain:

(1) the general classification scheme (i.e. "Data Input", "Model", "Post-processing" as categories) 
(2) the list of flavors available
(3) the convention behind the flavor names

This should go to the Python ML specific page and we can link it here.

Reply:
OK - I see that we do the above in the current file, but I got confused by having the Workflow/Subworkflow description at the top. Maybe let's move that further down in the page after we speak about the flavors?

The workflow you are explaining is just an example (default) workflow and we should not start the explanation with it.  We should explain the approach and the possibilities (flavors = "Lego" blocks, users can combine them in their preferred way), and then lay out the example workflow and conventions used there, maybe even in a separate page.

Reply:
Separated the workflows out into their own document, but made sure to include the "lego" analogy when discussing the flavors. Also added a section discussing the differing behavior based on whether a workflow is being run in "train" or "predict" mode.